### PR TITLE
Like and unlike buttons cannot be spammed now

### PR DESCRIPTION
**Pull request by Ilithor**

### CODE REVIEW

File: src/components/like/LikeButton.js [hunk 1]
```diff
@@ -21,9 +21,19 @@ import { getLikePost } from '../../redux/actions/dataActions';
  * @param {any} props.getLikePost
  */
 export const LikeButton = ({ postId, getLikePost }) => {
-  const likePost = () => getLikePost(postId);
+  const [disabled, setDisabled] = useState(false);
+  const handleDisabled = () => setDisabled({ ...disabled, disabled: true });
+  const handleEnabled = () => setDisabled({ ...disabled, disabled: false });
```
[the remark this answers is not in the document]
Comment (Ilithor) [reply src/components/like/LikeButton.js]: fixed

File: src/components/like/UnlikeButton.js [hunk 2]
```diff
@@ -21,10 +21,20 @@ import { getUnlikePost } from '../../redux/actions/dataActions';
  * @param {any} props.getUnlikePost
  */
 export const UnlikeButton = ({ postId, getUnlikePost }) => {
-  const unlikePost = () => getUnlikePost(postId);
+  const [disabled, setDisabled] = useState(false);
+  const handleDisabled = () => setDisabled({ ...disabled, disabled: true });
+  const handleEnabled = () => setDisabled({ ...disabled, disabled: false });
```
[the remark this answers is not in the document]
Comment (Ilithor) [reply src/components/like/UnlikeButton.js]: fixed

File: src/components/like/LikeButton.js [hunk 3]
```diff
@@ -21,9 +21,19 @@ import { getLikePost } from '../../redux/actions/dataActions';
...
+  const handleEnabled = () => setDisabled(false);
+  const likePost = () => {
+    getLikePost(postId);
+    handleEnabled();
+  };
   return (
-    <CustomButton tip='Like' onClick={likePost}>
+    <CustomButton
+      tip='Like'
+      onClick={(handleDisabled, likePost)}
```
Comment (jeffersoneagley) [inline src/components/like/LikeButton.js]: wait what.... you can do that? {(func1,func2)}???????????????????? /confuse

Comment (jeffersoneagley) [reply src/components/like/LikeButton.js]: apparently this doesn't actually work

File: src/util/CustomButton.js [hunk 4]
```diff
@@ -6,10 +6,23 @@ import Tooltip from '@material-ui/core/Tooltip';
...
+  tip,
+  btnClassName,
+  tipClassName,
+  disabled,
+}) => (
   <Tooltip title={tip} className={tipClassName} placement='top'>
-    <IconButton onClick={onClick} className={btnClassName}>
-      {children}
-    </IconButton>
+    <span>
```
Comment (jeffersoneagley) [inline src/util/CustomButton.js]: Why is there a span here?

Comment (Ilithor) [reply src/util/CustomButton.js]: Material-ui originally threw an error saying that it needs the span, but now I can't seem to repeat the error. Not sure if it's needed or not.

Comment (Ilithor) [reply src/util/CustomButton.js]: image

Nevermind, I got it to repeat. Without the span, it throws an error that the tooltip is confused by the disabled command so, there needs to be a span to circumnavigate the issue.

Comment (jeffersoneagley) [reply src/util/CustomButton.js]: Interesting, I guess this is a material-ui implementation specific thing.

**Pull request merged**